lib/oxide-barcode/src/lib.rs

```diff
@@ -102,39 +106,82 @@ impl VpdIdentity {
 mod tests {
     use super::*;
 
-    #[test]
-    fn parse_oxv1() {
-        let expected = VpdIdentity {
-            part_number: *b"123-0000456",
-            revision: 23,
-            serial: *b"TST01234567",
-        };
-
+    #[track_caller]
+    fn check_parse(input: &[u8], expected: VpdIdentity) {
         assert_eq!(
             expected,
-            VpdIdentity::parse(b"0XV1:1230000456:023:TST01234567").unwrap()
+            VpdIdentity::parse(input).unwrap(),
+            "parsing string: {}",
+            String::from_utf8_lossy(input),
         );
+
+        // We accept barcode strings that start with both leading zero and
+        // leading capital-O. Permute our input from one of these to the other
+        // to make sure both forms parse equivalently.
+        let mut copy = input.to_vec();
+        match copy[0] {
+            b'0' => copy[0] = b'O',
+            b'O' => copy[0] = b'0',
+            c => {
+                panic!("unexpected leading character in string: {}", c as char)
```

Comment:
nit: this wording feels slightly confusing, as the assertion a few lines above says "parsing string: {the input string}". this panic message kind of reads like the unexpected character was the _whole_ string, and we never print the whole string here. IMO, it might make more sense to say something like
```suggestion
                panic!("unexpected leading character {} in string: {}", c as char,, String::from_utf8_lossy(input))
```
or, just drop the "in string:" part if we don't think showing the whole string is important here (and it's probably not, since the `#[track_caller]` will make it clear which test failed...):
```suggestion
                panic!("unexpected leading character: {}", c as char)
```

Reply:
(this was supposed to be a review comment but GitHub seems to have eaten it somehow...)

Reply:
I've taken the latter diff, since this is checking correctness of test fixtures only.